### PR TITLE
Added explicit filter

### LINKED ISSUE #23: Toggle between explicit and non-explicit music
The response from the Spotify API contains data about whether certain tracks are explicit or not. The user should have a toggle switch that lets them toggle between Explicit and non-explicit.

Maybe the easiest way to do this is to generate 2 separate lists when recommendations are loaded in, 1 with all the tracks and 1 with only non-explicit tracks. Then when the user toggles we just show different tracks and send the selected list to Save as playlist.

Idea suggested by: RippedGeek from Reddit. Thanks Geek!

image

### PR DESCRIPTION
**Pull request by ad-dan**
Fixes #23

Works with the "Save Playlist" button as well as just displaying/hiding the songs.

### CODE REVIEW

**Comment by ad-dan** [comment]
Fixes #22

**Review by RyanRussell00 — CHANGES_REQUESTED** [review]
This works great! Please just make 2 minor changes:
- Pull latest changes from RyanRussell00:master and handle merge conflicts
- Reset the explicit checkbox if the user clicks on "Get Recommendations" again

**Comment by ad-dan** [comment]
I fixed the issues. Should I make a new PR or this PR is ok?

File: page.js [hunk 1]
```diff
@@ -49,256 +51,281 @@ const token_id = "personal-dj-token";
...
         if (responseIsSuccess(data)) {
-          trackResult = data.trackResult.tracks.items;
-          displaySearchResults(trackResult);
-        }
-        if (onDone && typeof onDone === "function") {
-          onDone(data);
+          recList_id = displayRecommendations(data.trackResult);
+          recList_cache = data.trackResult;
+
+          recList_id = displayRecommendations(data.trackResult);
```
Comment (RyanRussell00) [inline page.js:242]: Line #239 and #242 seem like they do the exact same thing, did you mean to set the filter_flag for one of them?

**Review by RyanRussell00 — COMMENTED** [review]
@ad-dan Please just address the above comment and then it should be good!

File: page.js (hunk 1 above)
Comment (ad-dan) [reply page.js:242]: No, It must be something leftover from the merge conflict. I will just remove it, seems like a silly mistake.

**Review by RyanRussell00 — COMMENTED** [review]
Looks great, thank you!

**Pull request merged**